hazelcast/src/main/java/com/hazelcast/internal/metrics/ProbeBuilder.java

```diff
@@ -34,10 +36,16 @@
      * @param source the object to pass to probeFn
      * @param metricName the value of "metric" tag
      * @param level the ProbeLevel
+     * @param unit the unit
      * @param probeFn the probe function
      * @throws NullPointerException if any of the arguments is null
      */
-    <S> void register(S source, String metricName, ProbeLevel level, DoubleProbeFunction<S> probeFn);
+    <S> void register(
```

Comment:
I guess for your usecase it makes sense to use this approach, but can't say I'm terribly excited about many arguments.

I would expect something like this

builder.withLevel(ProbeLevel.DEBUG).withProbeUnit(ProbeUnit.BYTES).register(foo,"foo");